cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/ElasticsearchConn.scala

```diff
@@ -0,0 +1,274 @@
+        tmp_json += ("count" -> JsNumber(count))
+        tmp_json += (joinSelectFieldString -> JsNumber(joinValue))
+        resArray = resArray.append(tmp_json)
+      }
+    }
+    else {
+      val buckets: Seq[JsObject] = (response \ "aggregations" \ groupAsList.head \ "buckets").get.as[Seq[JsObject]]
+      for (bucket <- buckets) {
+        val keyValue = if (bucket.keys.contains("key_as_string")) (bucket \ "key_as_string").get else (bucket \ "key").get
+        val jsLiquid = (bucket \ groupAsList.last \ "buckets").getOrElse(JsNull)
+        if (jsLiquid != JsNull) {
+          val liquid = jsLiquid.as[Seq[JsObject]]
+          for (drop <- liquid) {
+            var tmp_json = Json.obj()
+            tmp_json += (groupAsList.head -> keyValue)
+            tmp_json += (groupAsList.last -> JsString((drop \ "key_as_string").get.as[String]))
+            tmp_json += ("count" -> JsNumber((drop \ "doc_count").get.as[Int]))
+            resArray = resArray.append(tmp_json)
+          }
+        }
+        else {
+          var tmp_json = Json.obj()
+          tmp_json += (groupAsList.head -> keyValue)
+          tmp_json += ("count" -> JsNumber((bucket \ "doc_count").get.as[Int]))
+          resArray = resArray.append(tmp_json)
+        }
+      }
+    }
+    resArray
+  }
+
+  private def parseNonGroupByResponse(response: JsValue, jsonQuery: JsObject, aggregation: String): JsValue = {
+    aggregation match {
+      case "" => { // Search query without aggregation
+        val sourceJsValue = (response.asInstanceOf[JsObject] \ "hits" \ "hits").getOrElse(JsNull)
+        if (sourceJsValue != JsNull) {
+          val sourceArray = sourceJsValue.as[Seq[JsObject]]
+          if (jsonQuery.keys.contains("_source")) {
+            val returnArray = sourceArray.map(doc => parseSource(doc.value("_source").as[JsObject]))
+            return Json.toJson(returnArray)
+          }
+          val returnArray = sourceArray.map(doc => doc.value("_source"))
+          return Json.toJson(returnArray)
+        }
+        return Json.arr()
+      }
+      case "count" => {
+        val asField = (jsonQuery \ "aggregation" \ "as").get.toString().stripPrefix("\"").stripSuffix("\"")
+        val count = (response.asInstanceOf[JsObject] \ "hits" \ "total").get.as[JsNumber]
+        return Json.arr(Json.obj(asField -> count))
+      }
+      case "min" | "max" => {
+        val asField = (jsonQuery \ "aggregation" \ "as").get.toString().stripPrefix("\"").stripSuffix("\"")
+
+        if (response.as[JsObject].keys.nonEmpty) {
+          val min_obj = (response \ "aggregations" \ asField).as[JsObject]
+          val res = if (min_obj.keys.contains("value_as_string")) (min_obj \ "value_as_string").getOrElse(JsNull) else (min_obj \ "value").get
+          if (res != JsNull) {
+            val jsonObjRes = Json.obj(asField -> res)
+            return Json.arr(jsonObjRes)
+          }
+        }
+        return Json.arr()
+      }
+      case _ => ???
```

Comment:
If we'll not implement `SUM`, `AVG` in this PR, let's add one todo comment here.

Reply:
Ok, I added TODO to our code